// @mantine/dropzone@7.5.2/cjs/Dropzone.cjs
'use client';
'use strict';

var React = require('react');
var reactDropzoneEsm = require('react-dropzone-esm');
var core = require('@mantine/core');
var hooks = require('@mantine/hooks');
var Dropzone_context = require('./Dropzone.context.cjs');
var DropzoneStatus = require('./DropzoneStatus.cjs');
var Dropzone_module = require('./Dropzone.module.css.cjs');

function _interopDefault (e) { return e && e.__esModule ? e : { default: e }; }

var React__default = /*#__PURE__*/_interopDefault(React);

const defaultProps = {
  loading: false,
  multiple: true,
  maxSize: Infinity,
  autoFocus: false,
  activateOnClick: true,
  activateOnDrag: true,
  dragEventsBubbling: true,
  activateOnKeyboard: true,
  useFsAccessApi: true,
  variant: "light",
  rejectColor: "red"
};
const varsResolver = core.createVarsResolver(
  (theme, { radius, variant, acceptColor, rejectColor }) => {
    const acceptColors = theme.variantColorResolver({
      color: acceptColor || theme.primaryColor,
      theme,
      variant
    });
    const rejectColors = theme.variantColorResolver({
      color: rejectColor || "red",
      theme,
      variant
    });
    return {
      root: {
        "--dropzone-radius": core.getRadius(radius),
        "--dropzone-accept-color": acceptColors.color,
        "--dropzone-accept-bg": acceptColors.background,
        "--dropzone-reject-color": rejectColors.color,
        "--dropzone-reject-bg": rejectColors.background
      }
    };
  }
);
const Dropzone = core.factory((_props, ref) => {
  const props = core.useProps("Dropzone", defaultProps, _props);
  const {
    classNames,
    className,
    style,
    styles,
    unstyled,
    vars,
    radius,
    disabled,
    loading,
    multiple,
    maxSize,
    accept,
    children,
    onDropAny,
    onDrop,
    onReject,
    openRef,
    name,
    maxFiles,
    autoFocus,
    activateOnClick,
    activateOnDrag,
    dragEventsBubbling,
    activateOnKeyboard,
    onDragEnter,
    onDragLeave,
    onDragOver,
    onFileDialogCancel,
    onFileDialogOpen,
    preventDropOnDocument,
    useFsAccessApi,
    getFilesFromEvent,
    validator,
    rejectColor,
    acceptColor,
    enablePointerEvents,
    loaderProps,
    mod,
    ...others
  } = props;
  const getStyles = core.useStyles({
    name: "Dropzone",
    classes: Dropzone_module,
    props,
    className,
    style,
    classNames,
    styles,
    unstyled,
    vars,
    varsResolver
  });
  const { getRootProps, getInputProps, isDragAccept, isDragReject, open } = reactDropzoneEsm.useDropzone({
    onDrop: onDropAny,
    onDropAccepted: onDrop,
    onDropRejected: onReject,
    disabled: disabled || loading,
    accept: Array.isArray(accept) ? accept.reduce((r, key) => ({ ...r, [key]: [] }), {}) : accept,
    multiple,
    maxSize,
    maxFiles,
    autoFocus,
    noClick: !activateOnClick,
    noDrag: !activateOnDrag,
    noDragEventsBubbling: !dragEventsBubbling,
    noKeyboard: !activateOnKeyboard,
    onDragEnter,
    onDragLeave,
    onDragOver,
    onFileDialogCancel,
    onFileDialogOpen,
    preventDropOnDocument,
    useFsAccessApi,
    validator,
    ...getFilesFromEvent ? { getFilesFromEvent } : null
  });
  hooks.assignRef(openRef, open);
  const isIdle = !isDragAccept && !isDragReject;
  return /* @__PURE__ */ React__default.default.createElement(Dropzone_context.DropzoneProvider, { value: { accept: isDragAccept, reject: isDragReject, idle: isIdle } }, /* @__PURE__ */ React__default.default.createElement(
    core.Box,
    {
      ...getRootProps({ ref }),
      ...getStyles("root", { focusable: true }),
      ...others,
      mod: [
        {
          accept: isDragAccept,
          reject: isDragReject,
          idle: isIdle,
          loading,
          "activate-on-click": activateOnClick
        },
        mod
      ]
    },
    /* @__PURE__ */ React__default.default.createElement(
      core.LoadingOverlay,
      {
        visible: loading,
        overlayProps: { radius },
        unstyled,
        loaderProps
      }
    ),
    /* @__PURE__ */ React__default.default.createElement("input", { ...getInputProps(), name }),
    /* @__PURE__ */ React__default.default.createElement("div", { ...getStyles("inner"), "data-enable-pointer-events": enablePointerEvents || void 0 }, children)
  ));
});
Dropzone.classes = Dropzone_module;
Dropzone.displayName = "@mantine/dropzone/Dropzone";
Dropzone.Accept = DropzoneStatus.DropzoneAccept;
Dropzone.Idle = DropzoneStatus.DropzoneIdle;
Dropzone.Reject = DropzoneStatus.DropzoneReject;

exports.Dropzone = Dropzone;
